client/vscode/src/chatActivation.ts

```diff
@@ -207,3 +272,45 @@ function insertTextAtCursorPosition(text: string) {
 function isServerEvent(command: string) {
     return command.startsWith('aws/chat/') || command === 'telemetry/event'
 }
+
+async function decodeRequest<T>(request: string, key: Buffer): Promise<T> {
+    const result = await jose.jwtDecrypt(request, key, {
+        clockTolerance: 60, // Allow up to 60 seconds to account for clock differences
+        contentEncryptionAlgorithms: ['A256GCM'],
+        keyManagementAlgorithms: ['dir'],
+    })
+
+    if (!result.payload) {
+        throw new Error('JWT payload not found')
+    }
+    return result.payload as T
+}
+
+function isEncryptedMessage(message: string, algorithm: string, encoding: string) {
+    // Check if the message has five parts separated by periods
+    const parts = message.split('.')
```

Comment:
nit: could you add comment about used encryption mechanism here?

Reply:
Yeah, this function is pretty barebones right now. I will expand and explain this as followup